File: web/projects/portal/src/app/composer/gui/vs/event/copy-vs-objects-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../../../common/viewsheet-client/viewsheet-event";

/**
 * Event used to copy objects in the composer.
 */
export class CopyVSObjectEvent implements ViewsheetEvent {
   /**
    * List of object names to be copied.
    */
   public objects: string[];

   /**
    * Whether this object is copy or cut.
    */
   public cut: boolean;

   /**
    * The new x offset of the object.
    */
   public xOffset: number;

   /**
    * The new y offset of the object.
    */
   public yOffset: number;

   // offset relative to the first object.
   public relative;

   /**
    * Creates a new instance of <tt>RemoveVSObjectEvent</tt>.
    *
    * @param objects the names of the objects being copied.
    * @param isCut   whether the objects are being copied or pasted.
    */
   constructor(objects: string[], isCut: boolean, xOffset: number = 0, yOffset: number = 0,
               relative: boolean = false)
   {
      this.objects = objects;
      this.cut = isCut;
      this.xOffset = xOffset;
      this.yOffset = yOffset;
      this.relative = relative;
   }
}

File: core/src/main/java/inetsoft/web/composer/vs/event/CopyVSObjectsEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.vs.event;

/**
 * Class that encapsulates the parameters for opening a viewsheet.
 *
 * @since 12.3
 */
public class CopyVSObjectsEvent {
   /**
    * Gets the list of object names to copy.
    */
   public String[] getObjects() {
      return objects;
   }

   /**
    * Sets the list of object names to copy.
    */
   public void setObjects(String[] objects) {
      this.objects = objects;
   }

   /**
    * Gets whether this is a cut event.
    */
   public boolean isCut() {
      return cut;
   }

   /**
    * Sets whether this is a cut event.
    */
   public void setCut(boolean isCut) {
      this.cut = isCut;
   }

   /**
    * Gets the x offset of the object.
    */
   public int getxOffset() {
      return xOffset;
   }

   /**
    * Sets the x offset of the object.
    */
   public void setxOffset(int xOffset) {
      this.xOffset = xOffset;
   }

   /**
    * Gets the y offset of the object.
    */
   public int getyOffset() {
      return yOffset;
   }

   /**
    * Sets the yOffset of the object.
    */
   public void setyOffset(int yOffset) {
      this.yOffset = yOffset;
   }

   /**
    * Check if the offset is relative to the first assembly position.
    */
   public boolean isRelative() {
      return relative;
   }

   public void setRelative(boolean relative) {
      this.relative = relative;
   }

   @Override
   public String toString() {
      return "CopyVSObjectsEvent{" + "cut='" + cut + "\'}";
   }

   private String[] objects;
   private boolean cut;
   private int xOffset;
   private int yOffset;
   private boolean relative;
}
